internal/csi/core/lvm/lvm.go

```
@@ -366,13 +369,25 @@ func (l *LVM) EnsurePhysicalVolumes(ctx context.Context) ([]string, error) {
 			span.SetStatus(codes.Error, "failed to create physical volume")
 			span.RecordError(err)
 			recorder.Eventf(corev1.EventTypeWarning, provisioningPhysicalVolumeFailed, "Provisioning physical volume %s failed: %s", device, err)
-			return nil, fmt.Errorf("failed to create physical volume: %w", err)
+			createErrors = append(createErrors, fmt.Errorf("failed to create physical volume %s: %w", device, err))
+			continue
 		}
+		finalPvs = append(finalPvs, device)
 		recorder.Eventf(corev1.EventTypeNormal, provisionedPhysicalVolume, "Successfully provisioned physical volume %s", device)
 		log.V(1).Info("created physical volume", "device", device)
 	}
 
-	return devices, nil
+	if len(finalPvs) == 0 {
+		if len(createErrors) > 0 {
+			log.Error(fmt.Errorf("failed to create physical volumes for some disks: %v", createErrors), "failed to create physical volumes")
+			return nil, errors.Join(createErrors...)
+		}
+		log.Error(fmt.Errorf("no disks are available"), "no disks are available")
+		// return as resource exhausted to prompt scheduler to move the workload to another node
+		return nil, fmt.Errorf("%w: no disks are available", core.ErrResourceExhausted)
```

This only logs - do we need to return the error instead?